### PR TITLE
refactor: n_threads and n_target_partitions are non-zero

### PR DESCRIPTION
**Pull request by crepererum**
Zero values will just panic. Prevent that earlier.

### CODE REVIEW

File: iox_tests/src/catalog.rs [hunk 1]
```diff
@@ -37,13 +36,9 @@ use schema::{
     sort::{adjust_sort_key_columns, compute_sort_key, SortKey},
     Projection, Schema,
 };
-use std::{collections::HashMap, sync::Arc};
+use std::{collections::HashMap, num::NonZeroUsize, sync::Arc};
 use uuid::Uuid;
 
-/// Global executor used by all test catalogs.
```
Comment (crepererum) [inline iox_tests/src/catalog.rs:43]: Drive-by clean-up. `DedicatedExecutors::new_testing()` already uses a global test fixture under the hood, no need to create yet another one.

File: influxdb_iox/src/commands/run/compactor2.rs [hunk 2]
```diff
@@ -109,7 +110,10 @@ pub async fn command(config: Config) -> Result<(), Error> {
     let num_threads = config
         .compactor_config
         .query_exec_thread_count
-        .unwrap_or_else(|| num_cpus::get() - 1_usize);
+        .unwrap_or_else(|| {
+            NonZeroUsize::new(num_cpus::get().saturating_sub(1))
+                .unwrap_or_else(|| NonZeroUsize::new(1).unwrap())
+        });
```
Comment (crepererum) [inline influxdb_iox/src/commands/run/compactor2.rs:116]: This was actually problematic. We would have panicked in deployment scenarios w/ a single CPU.

File: influxdb_iox/src/commands/run/all_in_one.rs [hunk 3]
```diff
@@ -585,7 +585,8 @@ pub async fn command(config: Config) -> Result<()> {
 
     // TODO: make num_threads a parameter (other modes have it
     // configured by a command line)
-    let num_threads = num_cpus::get();
+    let num_threads = NonZeroUsize::new(num_cpus::get())
+        .unwrap_or_else(|| NonZeroUsize::new(1).expect("1 is valuid"));
```
Comment (carols10cents) [inline influxdb_iox/src/commands/run/all_in_one.rs:589]: Nit, typo:

```suggestion
        .unwrap_or_else(|| NonZeroUsize::new(1).expect("1 is valid"));
```

**Review by carols10cents — APPROVED** [review]
One tiny typo nit, otherwise looks great!
